hw/drivers/rtc/ds1307/src/ds1307.c

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright 2022 Jesus Ipanienko
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string.h>
+#include <assert.h>
+
+#include <os/mynewt.h>
+#include <hal/hal_i2c.h>
+#include <i2cn/i2cn.h>
+#include <modlog/modlog.h>
+#include <stats/stats.h>
+
+#include <ds1307/ds1307.h>
+
+/* Define stat names for querying */
+STATS_NAME_START(ds1307_stat_section)
+    STATS_NAME(ds1307_stat_section, read_count)
+    STATS_NAME(ds1307_stat_section, write_count)
+    STATS_NAME(ds1307_stat_section, read_errors)
+    STATS_NAME(ds1307_stat_section, write_errors)
+STATS_NAME_END(ds1307_stat_section)
+
+int ds1307_write_regs(struct ds1307_dev *ds1307, uint8_t addr, uint8_t *vals, uint8_t count)
+{
+    int rc;
+    uint8_t payload[count + 1];
+
+    payload[0] = addr;
+    memcpy(payload + 1, vals, count);
+
+    struct hal_i2c_master_data data_struct = {
+        .address = DS1307_I2C_ADDR,
+        .len = count + 1,
+        .buffer = payload,
+    };
+
+    STATS_INC(ds1307->stats, write_count);
+    rc = i2cn_master_write(ds1307->hw_cfg.i2c_num, &data_struct, OS_TICKS_PER_SEC / 10, 1, 1);
+    if (rc)
```

Comment:
mynewt uses `{` in same line as `if`, I don't know why CI did not detect this automatically @utzig